### ISSUE TITLE
Bug: Proxy Settings refusing to work correctly with http proxy

### ISSUE THREAD

**Issue by c00k3** [MISP/MISP#10005]
### Actual behavior

(Running MISP:latest on Docker) Proxy HTTP does not work with MISP. It fails and also breaks the Diagnostics page in https://<our_instance>.com/servers/serverSettings/diagnostics

My fix has been to amend the CurlClient.php file, located in /var/www/MISP/app/lib/Tools/ - declaring the option for:
`CURLOPT_PROXYPORT => '<the_port_we_use>'`

This I place within the following section:

private function generateDefaultOptions()
{
$options = [
CURLOPT_FOLLOWLOCATION => true // Allows to follow redirect
CURLOPT_MAXREDIRS => 10,
CURLOPT_RETURNTRANSFER => true, // Should cURL return or print out the data? (true = return, false = print)
CURLOPT_HEADER => false, // Include header in result?
CURLOPT_TIMEOUT => $this->timeout, // Timeout in seconds
CURLOPT_PROTOCOLS => CURLPROTO_HTTPS | CURLPROTO_HTTP, // be sure that only HTTP and HTTPS protocols are enabled
CURLOPT_PROXYPORT => '<the_port_we_use>'
];

Just wondering if this could be assessed because whenever the container is restarted or updated I have to run my script and pray that it still works to fix the bug.

### Expected behavior

PROXY: OK - Pull Events and Galaxies fine.

### Steps to reproduce

Restart docker containers or update MISP.

### Version

2.5.1

### Operating System

Ubuntu

### Operating System version

20.04

### PHP version

8.2.24

### Browser

Chrome, Safari

### Browser version

129.0.6668.101

### Relevant log output

```shell
I am currently unable to export any logs from our company machines, but working on getting some to upload here.
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by iglocska** [comment]
Weird, this should work. At the risk of revealing my lack of experience with the dockerised version: Do you set the proxy url yourself? `CURLOPT_PROXYPORT` is an optional parameter and can be set by passing the proxy port via the proxy url instead.

This is done via the line using CURLOPT_PROXY:

`$options[CURLOPT_PROXY] = "{$this->proxy['host']}:{$this->proxy['port']}";`

as described at https://curl.se/libcurl/c/CURLOPT_PROXY.html

I can see this becoming an issue if you've enabled curl partially then got locked out of the server settings page, but in that case I'd still just modify the MISP/app/Config/config.php file directly to resolve it, just make sure you update this block:

```
  'Proxy' => 
  array (
    'host' => '',
    'port' => 0,
    'method' => '',
    'user' => '',
    'password' => '',
  ),
```

**Comment by c00k3** [comment]
Yes, once this builds we set the proxy on the webUI in the settings under the proxy tab, with the host and port. This is reflected in the config.php

  'Proxy' => 
  array (
    'host' => 'http://proxy.company.com/‘,
    'port' => 3128,
    'method' => '',
    'user' => '',
    'password' => '',
  ),

Yet, this breaks in some way and doesn’t let the proxy connect until I amend the CurlConfig.php file. 

There are http_proxy and https_proxy env variables set for docker-compose to use and within the container it works but the application of MISP doesn’t consider those in the curl configuration.
